ops/pebble.py

```diff
@@ -1969,6 +1986,88 @@ def read1(self, n: int = -1) -> Union[str, bytes]:
         return self.read(n)
 
 
+class IdentityAccess(enum.Enum):
+    """Enum of identity access levels."""
+
+    ADMIN = 'admin'
+    READ = 'read'
+    METRICS = 'metrics'
+    UNTRUSTED = 'untrusted'
+
+
+@dataclasses.dataclass
+class LocalIdentity:
+    """Local identity configuration (for ucrednet/UID authentication)."""
+
+    user_id: int
+
+    @classmethod
+    def from_dict(cls, d: LocalIdentityDict) -> LocalIdentity:
+        """Create new LocalIdentity from dict parsed from JSON."""
+        return cls(user_id=d['user-id'])
+
+
+@dataclasses.dataclass
+class BasicIdentity:
+    """Basic identity configuration (for HTTP basic authentication)."""
+
+    password: str  # sha512-crypt-hashed password
+
+    @classmethod
+    def from_dict(cls, d: BasicIdentityDict) -> BasicIdentity:
+        """Create new BasicIdentity from dict parsed from JSON."""
+        return cls(password=d['password'])
+
+
+@dataclasses.dataclass
+class Identity:
+    """Pebble identity configuration."""
+
+    access: Union[IdentityAccess, str]
+    local: Optional[LocalIdentity] = None
+    basic: Optional[BasicIdentity] = None
+
+    def __post_init__(self) -> None:
+        """Validate that at least one of local or basic is provided."""
+        if self.local is None and self.basic is None:
+            raise ValueError('at least one of "local" or "basic" must be provided')
+
+    @classmethod
+    def from_dict(cls, d: IdentityDict) -> Identity:
+        """Create new Identity from dict parsed from JSON."""
+        if 'access' not in d:
+            raise ValueError('"access" key is required in IdentityDict')
+
+        try:
+            access = IdentityAccess(d['access'])
+        except ValueError:
+            access = d['access']
+
+        local = (
+            LocalIdentity.from_dict(d['local'])
+            if 'local' in d and d['local'] is not None
+            else None
+        )
+        basic = (
+            BasicIdentity.from_dict(d['basic'])
+            if 'basic' in d and d['basic'] is not None
+            else None
+        )
+
+        return cls(access=access, local=local, basic=basic)
+
+    def to_dict(self) -> IdentityDict:
+        """Convert this identity its dict representation."""
+        result: Dict[str, Any] = {
+            'access': self.access.value if isinstance(self.access, IdentityAccess) else self.access
```

Comment:
And without the union, and with the str-enum, this will just be `'access': self.access`